scheduler/models.py

```diff
@@ -383,6 +387,8 @@ def schedule(self):
         job = self.get_queue2().enqueue_at(
             scheduled_time,
             self.callable_func(),
+            on_success=callback_to_reschedule_cronjob,
+            on_failure=callback_to_reschedule_cronjob,
```

Comment:
I get why on_success, but why on_failure? There is a retry mechanism.

Also, I will write a test for it this time, but in the future, I would appreciate it if you can write tests for your changes.

Reply:
I wanted to be extra careful about it. The rescheduling has to happen no matter what.
Since we now depend too much on the scheduled Registry, i.e if the job doesn't end up being rescheduled once, the job, even enabled, will be in a limbo where no further executions will happen.

So, maybe (maybe!) the on_failure that will be retried and then succeed so the rescheduling will be done twice. (thinking it again, if any "schedule" action calls unschedule first, we might be fine if the job_id is the same). Worst case scenario, the job will ran more times. 
I have this code working, and overnight it did not generate even 1 failed job. and the scheduled column seems consistent.
Thank you for the tests, I will follow your lead on that.
Thanks so much for the attention to this! and for this project that you are truly taking it to the next level!